File: ruby/src/practitioner.rb
require './fhir_client'

class PractitionerClient
  def get_practitioner(id)
    FhirClient.new

    FHIR::Practitioner.read(id)
  end

  def search_practitioners_by_careteam_id(id)
    FhirClient.new

    params = {
      '_has:CareTeam:_id': id
    }
    FHIR::Practitioner.search(params)
  end

end

File: javascript/js/practitioner.js
import Client from "./client";

class PractitionerClient {
  async get() {
    const client = new Client();
    return client.request('Practitioner/1');
  }

  async searchByCareTeam() {
    const client = new Client();
    return client.request('Practitioner/?_has:CareTeam:_id=1');
  }
}

export default PractitionerClient;
